core.js

```diff
@@ -1609,28 +1609,31 @@ exokit.load = (src, options = {}) => {
   if (!url.parse(src).protocol) {
     src = 'http://' + src;
   }
-  return fetch(src)
-    .then(res => {
-      if (res.status >= 200 && res.status < 300) {
-        return res.text();
-      } else {
-        return Promise.reject(new Error('fetch got invalid status code: ' + res.status + ' : ' + src));
-      }
-    })
-    .then(htmlString => {
-      let baseUrl;
-      if (options.baseUrl) {
-        baseUrl = options.baseUrl;
-      } else {
-        baseUrl = utils._getBaseUrl(src);
-      }
+  return new Promise(resolve => {
```

Comment:
Making it so `.load()` returns a Promise which resolves the window. So when we call `.load()`, we can get a handle on the window to debug on.

Reply:
Hm, wasn't `return fetch(src)` already doing that? This is existing code:

![capture 4](https://user-images.githubusercontent.com/6926057/44294510-2ec7a580-a266-11e8-82be-3a87faa7b922.PNG)

```
(async () => { let p = require('./index').load('https://aframe.io/a-painter'); let o = await p; console.log(p.constructor.name, Object.keys(o)); })(); 1;
```

Reply:
It returned a Promise, it would resolve an XHR response, not the window object, which is more useful for debugging.

Reply:
I believe that was already how it worked. This is pre-PR.

```
$ node
(async () => {
  let p = require('./index').load('https://aframe.io/a-painter');
  let o = await p;
  console.log(o.constructor.name, typeof o.document.createElement); // window, not XHR or Response
})();
> Object function
```
The reason it works is the window is forwarded in the `.then` by design, because I had the same issue https://github.com/webmixedreality/exokit/blob/10ad1e89877215a9573ca8369d68027af86a1c48/core.js#L1628 :P.

Reply:
Ah, you're right. I removed that commit then.